src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -1134,6 +1134,14 @@ Namespace Microsoft.CodeAnalysis.Operations
             Return New LazyForEachLoopStatement(locals, loopControlVariable, collection, nextVariables, body, _semanticModel, syntax, type, constantValue, isImplicit)
         End Function
 
+        Private Function CreateBoundControlVariableOperation(boundForStatement As BoundForStatement) As IOperation
+            Dim localOpt = boundForStatement.DeclaredOrInferredLocalOpt
+            Dim controlVariable = boundForStatement.ControlVariable
+            Return If(localOpt IsNot Nothing AndAlso controlVariable?.Syntax.Kind = SyntaxKind.VariableDeclarator,
```

Comment:
>AndAlso controlVariable?.Syntax.Kind = SyntaxKind.VariableDeclarator [](start = 45, length = 68)

It feels like we should remove this condition, because an implicitly typed local should probably also be represented as variable declaration.